forc-plugins/forc-doc/src/render/item/context.rs

```diff
@@ -0,0 +1,354 @@
+                        })
+                        .collect();
+                    links.insert(BlockTitle::Fields, doc_links);
+                }
+                ContextType::EnumVariants(variants) => {
+                    let doc_links = variants
+                        .iter()
+                        .map(|variant| DocLink {
+                            name: variant.name.as_str().to_string(),
+                            module_info: ModuleInfo::from_ty_module(vec![], None),
+                            html_filename: format!("{}variant.{}", IDENTITY, variant.name.as_str()),
+                            preview_opt: None,
+                        })
+                        .collect();
+                    links.insert(BlockTitle::Variants, doc_links);
+                }
+                ContextType::RequiredMethods(methods) => {
+                    let doc_links = methods
+                        .iter()
+                        .map(|method| DocLink {
+                            name: method.name.as_str().to_string(),
+                            module_info: ModuleInfo::from_ty_module(vec![], None),
+                            html_filename: format!(
+                                "{}structfield.{}",
+                                IDENTITY,
+                                method.name.as_str()
+                            ),
+                            preview_opt: None,
+                        })
+                        .collect();
+                    links.insert(BlockTitle::RequiredMethods, doc_links);
+                }
+            }
+        }
+        DocLinks {
+            style: DocStyle::Item {
+                title: None,
+                name: None,
+            },
+            links,
+        }
+    }
+}
+impl Renderable for ItemContext {
+    fn render(self, render_plan: RenderPlan) -> Result<Box<dyn RenderBox>> {
+        let (title, rendered_list) = match self.context_opt {
+            Some(context) => {
+                let title = context.context_type.as_block_title();
+                let rendered_list = context.render(render_plan)?;
+                Ok((title, rendered_list))
+            }
+            None => Err(anyhow!(
+                "Safeguard against render call on empty context failed."
+            )),
+        }?;
+        let lct = title.html_title_string();
+        Ok(box_html! {
+            h2(id=&lct, class=format!("{} small-section-header", &lct)) {
+                : title.as_str();
+                a(class="anchor", href=format!("{IDENTITY}{lct}"));
+            }
+            : rendered_list;
+        })
+    }
+}
+
+#[derive(Clone, Debug)]
+pub(crate) enum ContextType {
+    /// structs
+    StructFields(Vec<TyStructField>),
+    /// storage
+    StorageFields(Vec<TyStorageField>),
+    /// enums
+    EnumVariants(Vec<TyEnumVariant>),
+    /// traits and abi, this can be split
+    /// at a later date if need be
+    RequiredMethods(Vec<TyTraitFn>),
```

Comment:
Unsure about how useful this documentation is. Perhaps remove / provide more context? Pun unintended

Reply:
Not sure why my brain thought this was enough info for someone else to understand what this does lol thanks